### PR TITLE
Fix: Configure logs early and APIs later

### PR DESCRIPTION
**Pull request by arthurpitman**
Currently logs are emitted before any logger formatting settings have ben applied. 

To fix this, this PR configures the logger as early as possible, and delays the initialization of the `configEndpoints` structure until it is first required, preventing at least this potential log message from being emitted too early.

As a side effect, the logger is re-initialized later with logging also being directed to `.log` and `-errors.log` files.

### CODE REVIEW

**Review by warber — APPROVED** [review]
pls write a small PR description :)

**Comment by UnseenWizzard** [comment]
Did you test what (if any) impact these changes have in terms of memory footprint? Without more than a cursory look, it looks like quite a few previously reused variables for API endpoint will now be allocated for each call to NewApis.

And is there any negative impact of setting up the full logger twice?

**Comment by arthurpitman** [comment]
> Did you test what (if any) impact these changes have in terms of memory footprint? Without more than a cursory look, it looks like quite a few previously reused variables for API endpoint will now be allocated for each call to NewApis.

Changed this to cache the endpoints after first request which should mitigate any concerns here.

> And is there any negative impact of setting up the full logger twice?

I don't believe so.

File: pkg/deploy/internal/classic/validation.go [hunk 1]
```diff
@@ -29,9 +29,16 @@ type (
 )
 
 type Validator struct {
+	apis        api.APIs
```
[the remark this answers is not in the document]
Comment (UnseenWizzard) [reply pkg/deploy/internal/classic/validation.go:33]: Actually just saw that this implements and is accessed via an interface anyway. So in just unexporting the `validator` struct works and removes any risk of using it wrong

**Pull request merged**